---
title: "Microbiome simulation"
subtitle: "Benchmarking methods for differential abundance testing"
author: "Sven Kleine Bardenhorst"
institute: "Institute for Epidemiology and Social Medicine"
date: "2020-09-18 (updated: `r Sys.Date()`)"
output:
  xaringan::moon_reader:
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = FALSE,
	message = FALSE,
	warning = FALSE
)
options(htmltools.dir.version = FALSE)
library(knitr)
library(phyloseq)
library(dplyr)
```

# Background

---

# Microbiome sequencing data
![16s gene](16s_var_region.png)

.pull-left[
- 16s rRNA gene sequencing
- shotgun metagenomic sequencing
- test
]

.pull-right[
![tree](tree.png)
]



---

# Microbiome data example
```{r}
data <- readRDS("example_data.RDS")

d <- data %>%
  filter(author=="Weis et al., 2019") %>%
  dplyr::select(-c(1:3)) %>%
  dplyr::select_if(function(x) sum(x)!=0)

library(DT)
d %>%
  DT::datatable(extensions = 'FixedColumns',
  options = list(
  dom = 't',
  scrollX = TRUE,
  scrollY = TRUE,
  scrollCollapse = TRUE))
  

```
---


---


# Data characteristics

- High-dimensional
- Compositional
- Zero-inflated
- Multivariate
- Unequal sequencing depth
- Over-dispersed

---

# Aims

- Provide pratical advice for choice of method
- Evalute methods with respect to
  -- different data characteristics
  -- different taxonomic levels


---

# Data-generating mechanisms

- true data-generating mechanism unknown
- simulation based on parametric model may not represent true data structure
- possible solution:
      - resampling from a large set of samples taken from healthy subjects
      - induce effects on top of resampling dataset using parametric model
- drawbacks:
      - "spurious" effects due to resampling
      - effect of interest will be a combinations of "true" effect and noise in the data
- Data characteristics may vary with taxonomic depth
      - lower ranks:
            - dimensionality
            - zero-inflation
            - over-dispersion
      - higher ranks:
            - compositionality
      
---

# Data generating mechanisms
- $n_{obs} = 50,100,150,200,400,1000$
- 


---

# Estimands / Effects in microbiome data

---

# Methods

---

# Performance measures

- Coverage  $$Pr(\hat{\theta}_{low} < \theta < \hat{\theta}_{upp})$$
- Power / type-I error rate $$ Pr(p_i \leq \alpha)$$
- Bias $$ E[\hat{\theta}] - \theta $$
- Empirical SE $$ \sqrt{Var(\hat{\theta})} $$
- MSE $$ E[(\hat{\theta} - \theta)^2] $$


---
